packages/edit-post/src/components/layout/index.native.js

```diff
@@ -83,6 +92,19 @@ class Layout extends Component {
 		this.setState( { rootViewHeight: height }, sendNativeEditorDidLayout );
 	}
 
+	shouldShowTooltip() {
+		requestStarterPageTemplatesTooltipShown( ( tooltipShown ) => {
+			if ( ! tooltipShown ) {
+				this.setState( { tooltipVisible: true } );
+				setStarterPageTemplatesTooltipShown( true );
+			}
+		} );
+	}
+
+	onTooltipHidden() {
+		this.setState( { tooltipVisible: false } );
+	}
+
```

Comment:
I'm wondering why do we need all these Tooltip logic here, maybe you explained it somewhere you can link me to? Thanks!

Reply:
> Regarding the Tooltip, I have a pending to be merged [PR](https://github.com/WordPress/gutenberg/pull/27640) that removes the `PageTemplatePicker` and its tooltip. I remember we talked about trying to have a common use for both tooltips but since it will only be used in the Focal picker, we can just have its own implementation there, what do you think?

@geriux the point may be moot now due to your own PR referenced above.

>I'm wondering why do we need all these Tooltip logic here, maybe you explained it somewhere you can link me to? Thanks!

To answer your question directly, we needed to relocate the logic to support creating a `Tooltip` component that [supported both use cases](https://github.com/WordPress/gutenberg/pull/25810#issuecomment-710587198), both the [page template picker](https://github.com/dcalhoun/gutenberg/commit/a6526f2dc52809ade95703df67a42f31d29836ac) and focal point picker. Supporting both use cases definitely added a lot of complexity, [which I didn't believe was worth the cost](https://github.com/WordPress/gutenberg/pull/25810#issuecomment-711396858).

Given that the tooltip is being removed from the page template picker, my recommendation would be that we revert the work I did to support both use cases. That would allow us to simplify the `Tooltip` component a good bit.

What do you think about simplifying the tooltip now that the page template picker no longer needs it?

Reply:
>What do you think about simplifying the tooltip now that the page template picker no longer needs it?

That would be perfect! Thanks!